3p/ampcontext.js

```diff
@@ -62,24 +120,33 @@ export class AmpContext {
    *    every time we receive a page visibility message.
    */
   observePageVisibility(callback) {
-    return this.client_.makeRequest(
-        MessageType.SEND_EMBED_STATE,
-        MessageType.EMBED_STATE,
-        callback);
+    return this.visibilityObservable_.add(callback);
   };
 
   /**
    *  Send message to runtime to start sending intersection messages.
-   *  @param {function(Object)} callback Function to call every time we receive
-   *  an intersection message.
+   *  @param {function(Array<Object>)} callback Function to call every time we
+   *    receive an intersection message.
    *  @returns {function()} that when called stops triggering the callback
    *    every time we receive an intersection message.
    */
   observeIntersection(callback) {
-    return this.client_.makeRequest(
+    const unlisten = this.client_.makeRequest(
```

Comment:
This is odd, why does the implementation for `IframeMessagingClient` override callbacks?

Added an observable for now, but do you think it would make more sense to either do:

A. Modify `IframeMessagingClient.makeRequest` so that the requests don't override
B. Add a method in `IframeMessagingClient` that would *add* callbacks instead of overriding them?

Any of the options will avoid littering `AmpContext` with observables.